libc/string_errsig.c

```diff
@@ -0,0 +1,251 @@
+		TEST_ASSERT_NOT_NULL(strsignal(signals_codes[i]));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strsignal(signals_codes[i])));
+	}
+}
+
+
+TEST(string_errsing, strsignal_invaild)
+{
+	static const int values[] = { -64, -1, 0, 128, 1024 };
+	for (int i = 0; i < (sizeof values / sizeof *values); i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strstr(strsignal(values[i]), "Unknown signal"));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsing, strsignal_message)
+{
+#ifdef __phoenix__
+	TEST_ASSERT_EQUAL_STRING("Information request", strsignal(SIGINFO));
+	TEST_ASSERT_EQUAL_STRING("EMT trap", strsignal(SIGEMT));
+#endif
+
+	TEST_ASSERT_EQUAL_STRING("Hangup", strsignal(SIGHUP));
+	TEST_ASSERT_EQUAL_STRING("Interrupt", strsignal(SIGINT));
+	TEST_ASSERT_EQUAL_STRING("Quit", strsignal(SIGQUIT));
+	TEST_ASSERT_EQUAL_STRING("Illegal instruction", strsignal(SIGILL));
+	TEST_ASSERT_EQUAL_STRING("Trace/breakpoint trap", strsignal(SIGTRAP));
+	TEST_ASSERT_EQUAL_STRING("Aborted", strsignal(SIGABRT));
+	TEST_ASSERT_EQUAL_STRING("Floating point exception", strsignal(SIGFPE));
+	TEST_ASSERT_EQUAL_STRING("Killed", strsignal(SIGKILL));
+	TEST_ASSERT_EQUAL_STRING("Bus error", strsignal(SIGBUS));
+	TEST_ASSERT_EQUAL_STRING("Segmentation fault", strsignal(SIGSEGV));
+	TEST_ASSERT_EQUAL_STRING("Bad system call", strsignal(SIGSYS));
+	TEST_ASSERT_EQUAL_STRING("Broken pipe", strsignal(SIGPIPE));
+	TEST_ASSERT_EQUAL_STRING("Alarm clock", strsignal(SIGALRM));
+	TEST_ASSERT_EQUAL_STRING("Terminated", strsignal(SIGTERM));
+	TEST_ASSERT_EQUAL_STRING("Urgent I/O condition", strsignal(SIGURG));
+	TEST_ASSERT_EQUAL_STRING("Stopped (signal)", strsignal(SIGSTOP));
+	TEST_ASSERT_EQUAL_STRING("Stopped", strsignal(SIGTSTP));
+	TEST_ASSERT_EQUAL_STRING("Continued", strsignal(SIGCONT));
+	TEST_ASSERT_EQUAL_STRING("Child exited", strsignal(SIGCHLD));
+	TEST_ASSERT_EQUAL_STRING("Stopped (tty input)", strsignal(SIGTTIN));
+	TEST_ASSERT_EQUAL_STRING("Stopped (tty output)", strsignal(SIGTTOU));
+	TEST_ASSERT_EQUAL_STRING("I/O possible", strsignal(SIGIO));
+	TEST_ASSERT_EQUAL_STRING("CPU time limit exceeded", strsignal(SIGXCPU));
+	TEST_ASSERT_EQUAL_STRING("File size limit exceeded", strsignal(SIGXFSZ));
+	TEST_ASSERT_EQUAL_STRING("Virtual timer expired", strsignal(SIGVTALRM));
+	TEST_ASSERT_EQUAL_STRING("Profiling timer expired", strsignal(SIGPROF));
+	TEST_ASSERT_EQUAL_STRING("Window changed", strsignal(SIGWINCH));
+	TEST_ASSERT_EQUAL_STRING("User defined signal 1", strsignal(SIGUSR1));
+	TEST_ASSERT_EQUAL_STRING("User defined signal 2", strsignal(SIGUSR2));
+}
+
+
+TEST(string_errsing, strsignal_real_time)
+{
+/* sigrtmin not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	TEST_ASSERT_EQUAL_STRING("Real-time signal 0", strsignal(SIGRTMIN));
+	TEST_ASSERT_EQUAL_STRING("Real-time signal 1", strsignal(SIGRTMIN + 1));
+	TEST_ASSERT_EQUAL_STRING("Real-time signal 16", strsignal(SIGRTMIN + 16));
+#endif
+}
```

Comment:
As it's described in doc:
"It shall use the same set of messages as the `psignal()` function"

I know that it wasn't planned, but we can also add `psignal()` tests there and cover this sentence.